plugins/modules/nmcli.py

```diff
@@ -2384,6 +2384,25 @@ def down_connection(self):
         cmd = [self.nmcli_bin, "con", "down", self.conn_name]
         return self.execute_command(cmd)
 
+    def get_connection_state(self):
+        """Get the current state of the connection"""
+        cmd = [self.nmcli_bin, "--terse", "--fields", "GENERAL.STATE", "con", "show", self.conn_name]
+        (rc, out, err) = self.execute_command(cmd)
+        if rc != 0:
+            raise NmcliModuleError(err)
+
+        lines = [ll.strip() for ll in out.splitlines() if "GENERAL.STATE" in ll]
+        for line in lines:
+            if "GENERAL.STATE" in line:
```

Comment:
Since `lines` only contains items that have `GENERAL.STATE` in them (line 2394), this `if` (line 2396) is redundant.

Moreover, since it returns on the very first element, this entire loop is redundant, right? You can just return None if the list is empty or return the splitted value of the first item of `lines`.